ABC-order variant analysis linting config

dfarrell07 opened the pull request:
This job was called vulnerability variant analysis during most of
development. It should have been reordered when the name was shortened.

Relates-to: submariner-io/submariner#1970
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt commented on .github/workflows/linting.yml:
```diff
@@ -116,6 +116,24 @@ jobs:
         run: make packagedoc-lint
 
+  variant-analysis:
+    name: Variant Analysis
```
Wouldn’t it be better to give this job a self-explanatory name? As it stands, it only indicates that it’s a variant of some other job. Wouldn’t it be better to name it “Vulnerability Scanning (CodeQL)” (and rename the Anchore job accordingly)?

dfarrell07 replied:
"Variant" in this case isn't meant as "variant of some other job", but instead as the name of a category of static analysis.

> Variant analysis is the process of using a known security vulnerability as a seed to find similar problems in your code. It’s a technique that security engineers use to identify potential vulnerabilities, and ensure these threats are properly fixed across multiple codebases.
Querying code using CodeQL is the most efficient way to perform variant analysis. You can use the standard CodeQL queries to identify seed vulnerabilities, or find new vulnerabilities by writing your own custom CodeQL queries. Then, develop or iterate over the query to automatically find logical variants of the same bug that could be missed using traditional manual techniques.

https://codeql.github.com/docs/codeql-overview/about-codeql/

CodeQL doesn't only do variant analysis for security issues, it also has semantic queries/rules for other types of issues.

https://github.com/github/codeql/tree/main/go/ql/src

I dropped "vulnerability" from the description because I don't think any of the [issues it found in our codebase](https://github.com/submariner-io/submariner/issues/1970#issue-1343160083) were security-related.

I can add some more docs like this to the commit messages?

skitt replied:
Ah right, TIL, thanks! I hadn’t found anything relevant in a web search, so it might be worth at least linking to the “about CodeQL” page in the commit message.

dfarrell07 replied:
Yeah there aren't a lot of docs, I'll spread some more info around.

dfarrell07 replied:
Added to the commit message :heavy_check_mark:

tpantelis commented:
It looks like the System Tests were broken by https://github.com/submariner-io/shipyard/pull/924 @mkolesnik

dfarrell07 commented:
> It looks like the System Tests were broken by [submariner-io/shipyard#924](https://github.com/submariner-io/shipyard/pull/924) @mkolesnik

Cross-linking [my comment about adding System Tests to the Consuming Projects tests](https://github.com/submariner-io/shipyard/pull/924#issuecomment-1233061460).

dfarrell07 committed:
ABC-order variant analysis linting config

This job was called vulnerability variant analysis during most of
development. It should have been reordered when the name was shortened.

https://codeql.github.com/docs/codeql-overview/about-codeql/

Relates-to: submariner-io/submariner#1970
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
The rebase here keeps getting broken by updates to the codeql-action GHA SHA.
